tuf/data/roles.go

```diff
@@ -101,6 +113,111 @@ type Role struct {
 	Email            string   `json:"email,omitempty"`
 }
 
+// BaseRole is an internal representation of a root/targets/snapshot/timestamp role, with its public keys included
+type BaseRole struct {
+	Keys      map[string]PublicKey `json:"keys"`
+	Name      string               `json:"name"`
+	Threshold int                  `json:"threshold"`
+}
+
+// ListKeys retrieves the public keys valid for this role
+func (b BaseRole) ListKeys() KeyList {
```

Comment:
`KeyList` exists purely for deserialization purposes because `PublicKey` is an interface so we can't deserialize directly to it. Instead we can define a custom unmarshaller on `KeyList` that is able to translate each incoming key into the correct type and assign it into the underlying `[]PublicKey`